src/main/java/app/PostRestClient.java

```diff
@@ -0,0 +1,77 @@
+package app;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+import java.util.Map;
+
+public class PostRestClient {
+
+    private final RestTemplate restTemplate;
+
+    public PostRestClient() {
+        this.restTemplate = new RestTemplate();
```

Comment:
you should pass it in costructor - new instance should be created in Configuration class